autopush/websocket.py

```diff
@@ -270,21 +270,26 @@ def process_hello(self, data):
         self.uaid = uaid
 
         connect = data.get("connect")
-        if connect and self.ap_settings.pinger:
+        if connect and self.ap_settings.bridge:
             self.transport.pauseProducing()
-            d = deferToThread(self.ap_settings.pinger.register, uaid, connect)
+            d = deferToThread(self.ap_settings.bridge.register, uaid, connect)
             d.addCallback(self._check_router, True)
             d.addErrback(self.err_hello)
         else:
             self._check_router(False)
 
-    def _check_router(self, paused=False):
+    def _check_router(self, paused=False, bridge_register=None):
         if paused:
             self.transport.resumeProducing()
+        if bridge_register is not None and bridge_register is False:
```

Comment:
My preference is to be explicit. If the bridge registration fails, the client can retry without bridge registration if so desired. Otherwise we risk the potential of "eating" the bridge registration failure and the client falsely believing that it can break the web-socket connection.